src/core/model/Compass.cpp

```diff
@@ -24,7 +24,7 @@ auto Compass::getToolRange(bool transformed) const -> Range {
         rg.addPoint(-h, -h);
         rg.addPoint(h, h);
     }
-    rg.addPadding(.5 * xoj::view::CompassView::LINE_WIDTH_IN_CM * CM);  // account for line width
+    rg.addPadding(2 + .5 * xoj::view::CompassView::LINE_WIDTH_IN_CM * CM);  // account for line width
```

Comment:
I would suggest to replace that number `2` by a constexpr as well and add a comment mentioning the digits and marks that should not be cut off.